--- dillo-0.6.6/src/interface.c.orig	2006-10-21 00:38:41.000000000 +0200
+++ dillo-0.6.6/src/interface.c	2006-10-21 00:39:25.000000000 +0200
@@ -598,7 +598,7 @@
    gtk_window_add_accel_group(GTK_WINDOW(bw->main_window), bw->accel_group);
 
    /* set window title */
-   g_snprintf(buf, 64, "Version %s", VERSION);
+   g_snprintf(buf, 64, "Dillo Version %s", VERSION);
    a_Interface_set_page_title(bw, buf);
 
    box1 = gtk_vbox_new(FALSE, 0);
@@ -804,19 +804,14 @@
 }
 
 /*
- * Set the title of the browser window to start with "Dillo: "
- * prepended to it.
+ * Set the title of the browser window
+ * "Dillo: " prefix removed in compliance with modern standards
  */
 void a_Interface_set_page_title(BrowserWindow *bw, char *title)
 {
-   GString *buf;
-
    g_return_if_fail (bw != NULL && title != NULL);
 
-   buf = g_string_new("");
-   g_string_sprintfa(buf, "Dillo: %s", title);
-   gtk_window_set_title(GTK_WINDOW(bw->main_window), buf->str);
-   g_string_free(buf, TRUE);
+   gtk_window_set_title(GTK_WINDOW(bw->main_window), title);
 }
 
 /*
