--- module-init-tools-3.2.2/depmod.c	2006-10-09 20:07:43.000000000 +0200
+++ module-init-tools-3.2.2/depmod.c	2006-10-09 20:00:49.000000000 +0200
@@ -216,11 +216,7 @@
 	char pathname[strlen(argv[0])+1];
 	char oldname[strlen("depmod") + strlen(argv[0]) + sizeof(".24")];
 
-	memset(pathname, 0, strlen(argv[0])+1);
-	sep = strrchr(argv[0], '/');
-	if (sep)
-		memcpy(pathname, argv[0], sep - argv[0]+1);
-	sprintf(oldname, "%s%s.24", pathname, "depmod");
+	sprintf(oldname, "%s.24", argv[0]);
 
 	/* Recursion detection: we need an env var since we can't
 	   change argv[0] (as older modutils uses it to determine
