--- ../mplayer-1.0pre7-r2/MPlayer-1.0pre7/etc/codecs.conf	2005-04-10 18:44:29.000000000 +0200
+++ MPlayer-1.0pre7/etc/codecs.conf	2006-09-21 20:21:20.000000000 +0200
@@ -2309,16 +2309,6 @@
   driver ffmpeg
   dll "mp3on4"
 
-audiocodec ffmp3
-  info "FFmpeg MPEG layer-3 audio decoder"
-  comment "integer only"
-  status working
-  format 0x55
-  format 0x33706d2e  ; ".mp3" CBR/VBR MP3 (MOV files)
-  format 0x5500736d  ; "ms\0\x55" older mp3 fcc (MOV files)
-  driver ffmpeg
-  dll "mp3"
-
 audiocodec ffmp3adu
   info "FFmpeg MPEG layer-3 adu audio decoder"
   comment "integer only"
@@ -2327,14 +2317,6 @@
   driver ffmpeg
   dll "mp3adu"
 
-audiocodec ffmp2
-  info "FFmpeg MPEG layer-1 and layer-2 audio decoder"
-  comment "integer only"
-  status working
-  format 0x50
-  driver ffmpeg
-  dll "mp2"
-
 audiocodec mad
   info "libMAD MPEG layer 1-2-3"
   status working
